scripts/wrapper/withdraw.js

```diff
@@ -48,25 +77,50 @@ module.exports = function (web3, artifacts) {
 
     assertGoodArguments(argv)
 
-    let withdrawals = JSON.parse(await fs.readFile(argv.withdrawalFile, "utf8"))
-    const tokenInfoPromises = fetchTokenInfoForFlux(withdrawals)
-    const exchange = await getExchange(web3)
+    let withdrawals
+    let tokenInfoPromises
+    if (argv.withdrawalFile) {
+      withdrawals = JSON.parse(await fs.readFile(argv.withdrawalFile, "utf8"))
+      tokenInfoPromises = fetchTokenInfoForFlux(withdrawals)
+    } else {
+      const exchangePromise = getExchange(web3)
+      if (argv.tokens) tokenInfoPromises = fetchTokenInfoAtAddresses(argv.tokens)
+      else {
+        const tokenAddresses = await Promise.all(
+          argv.tokenIds.map(async (id) => (await exchangePromise).tokenIdToAddressMap(id))
+        )
+        tokenInfoPromises = fetchTokenInfoAtAddresses(tokenAddresses)
+      }
+      const exchange = await exchangePromise
```

Comment:
I don't understand the purpose of creating the promise if it is only going to be awaited a couple of lines below. This seems like an unnecessary line of code when we can simple write `const exchange = await getExchange(web3)` back at line 86.

Reply:
I'd argue in favor of the current code: it reduces the running time by one network request assuming that `argv.tokens` is given by the user.
This is because (if `argv.token` is set) the fetching of the exchange is executed in parallel to the fetching of token information, while if you await the exchange in line 86 then the fetching of token information only starts immediately after the exchange has been retrieved.

Reply:
Ok then. If this is the case, then I am in favour of this. Was not aware this how it worked! Cool stuff.